src/main/java/org/fogbowcloud/arrebol/core/models/job/Job.java

```diff
@@ -2,45 +2,56 @@
 
 import org.apache.log4j.Logger;
 import org.fogbowcloud.arrebol.core.models.task.Task;
+
+import javax.persistence.*;
 import java.io.Serializable;
 import java.util.*;
 
+@Entity
 public class Job implements Serializable {
 
 	private static final long serialVersionUID = -6111900503095749695L;
 	private static final Logger LOGGER = Logger.getLogger(Job.class);
 
-	private final String id;
-	private final String label;
+	@Id
+	private String id;
+	private String label;
+
+	@Enumerated(EnumType.STRING)
 	private JobState jobState;
 
+	@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+	@MapKey(name = "id")
 	private Map<String, Task> tasks;
 
-	public Job(String label){
+	public Job(String label, Map<String, Task> tasks){
 		this.id = UUID.randomUUID().toString();
 		this.jobState = JobState.SUBMITTED;
 		this.label = label;
-		this.tasks = new HashMap<String, Task>();
+		this.tasks = tasks;
 	}
 
-	public Job(String label, Map<String, Task> tasks){
-		this(label);
-		this.tasks = tasks;
+	Job(){
+		//Default constructor
 	}
 
-	public void addTask(Task task){
-		tasks.put(task.getId(), task);
+	public String getId(){
+		return this.id;
 	}
 
-	public Map<String, Task> getTasks(){
-		return this.tasks;
+	public String getLabel(){
+		return this.label;
 	}
 
-	public String getId(){
-		return this.id;
+	public JobState getJobState(){
+		return this.jobState;
 	}
 
 	public void setJobState(JobState jobState){
 		this.jobState = jobState;
 	}
+
+	public Map<String, Task> getTasks(){
```

Comment:
i'd be better to return a copy of this map

Reply:
Done.